core/da/dummy_client.go

```diff
@@ -107,3 +107,8 @@ func (c *DummyClient) RetrieveHeaders(ctx context.Context, dataLayerHeight uint6
 		Headers: headers,
 	}
 }
+
+// GetDA returns the underlying DA implementation
+func (c *DummyClient) GetDA() DA {
```

Comment:
this seems to defeat the purpose of the interface separation why is it needed? maybe there is a misstep in the design that we require raw DA being passed instead of the client